### PR TITLE
define problem marker

### PR DESCRIPTION
**Pull request by elharo**
@briandealwis

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.whitelist/plugin.xml [hunk 1]
```diff
@@ -12,4 +13,14 @@
          requiredSourceLevel="1.5">
      </compilationParticipant>
   </extension>
+  
+  <extension 
+      id="blacklistedClass" 
+      name="App Engine Whitelist Marker"
+      point="org.eclipse.core.resources.problemmarkers">
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.whitelist/plugin.xml]: This doesn't look right: the extension point is o.e.c.r.markers.

Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.whitelist/plugin.xml]: done. I wonder, is there a way to test that the extension points exist?

Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.whitelist/plugin.xml]: You could verify that the extension registry has > 0 extensions for the given extension point.

```
String definingBundleId = FrameworkUtil.getBundle(someClassInBundle).getSymbolicName();
IConfigurationElement[] elements = RegistryFactory.getRegistry().getConfigurationElementsFor(definingBundleId, extPointId);
assertThat(elements.length > 0);
```

Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.whitelist/plugin.xml]: On Tue, Apr 12, 2016 at 2:55 PM, Brian de Alwis <notifications@github.com>
wrote:

> You could verify that the extension registry has > 0 extensions for the
> given extension point.
>
> String definingBundleId = FrameworkUtil.getBundle(someClassInBundle).getSymbolicName();
> IConfigurationElement[] elements = RegistryFactory.getRegistry().getConfigurationElementsFor(definingBundleId, extPointId);
> assertThat(elements.length > 0);
>
>
Where does someClassInBundle come from?


-- 
Elliotte Rusty Harold
elharo@macfaq.com

Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.whitelist/plugin.xml]: You could embed the bundle symbolic name directly.  By referencing a class in that bundle, you insulate the test in the face of bundle renames.

Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.whitelist/plugin.xml]: On Tue, Apr 12, 2016 at 4:24 PM, Brian de Alwis <notifications@github.com>
wrote:

> You could embed the bundle symbolic name directly. By referencing a class
> in that bundle, you insulate the test in the face of bundle renames.


I think we're operating at a different levels here. Which string is this
bundle symbolic name you refer to?


-- 
Elliotte Rusty Harold
elharo@macfaq.com

Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.whitelist/plugin.xml]: The bundle symbolic name (BSN) is the human-readable name for a bundle.  In your case, `com.google.cloud.tools.eclipse.appengine.whitelist`.

`FrameworkUtil` is a helper class from OSGi that will find the bundle that contains a class.  The Eclipse Extension Registry tracks the extensions contributed by a bundle; the extension registry is supposed to be independent of OSGi, so it uses terms like contributors (= bundle) that have names ('namespaces' = BSN).

File: plugins/com.google.cloud.tools.eclipse.appengine.whitelist.test/src/com/google/cloud/tools/eclipse/appengine/whitelist/XmlTest.java [hunk 2]
```diff
@@ -50,4 +51,15 @@ public void testLoadExtensionClass() {
     }
   }
   
+  @Test
+  public void testLoadExtensionElements() {
+    NodeList extensions = document.getDocumentElement().getElementsByTagName("extension");
+    for (int i = 0; i < extensions.getLength(); i++) {
+      Element extension = (Element) extensions.item(i);
+      String point = extension.getAttribute("point");
+      Assert.assertTrue(point.length() > 0);
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.whitelist.test/src/com/google/cloud/tools/eclipse/appengine/whitelist/XmlTest.java]: Do you have a preference against static imports on Assert?
Ah, we can test the existence of the extension point with:
```
IExtensionRegistry registry = RegistryFactory.getRegistry();
IExtensionPoint extensionPoint = registry.getExtensionPoint(point);
Assert.assertNotNull(extensionPoint);
```

Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.whitelist.test/src/com/google/cloud/tools/eclipse/appengine/whitelist/XmlTest.java]: There's no rule at Google about static imports, one way or the other, but personally I like to know wyhere my methods are coming from. :-)

Added the test that we can load extensions points.

**Pull request merged**